jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/HostPort.java

```diff
@@ -176,10 +176,10 @@ else if (!isValidHostName(host))
 
                         port = parsePort(authority.substring(c + 1), unsafe);
                         // horribly bad port during unsafe
-                        if (unsafe && (port == BAD_PORT))
+                        if (unsafe && (port == UNDEFINED_PORT))
                         {
                             host = authority; // whole authority (no substring)
-                            port = -1;
+                            port = UNDEFINED_PORT;
                         }
```

Comment:
This code looks strange.   Why check == UNDEFINED and then set UNDEFINED?

Reply:
I agree, but I didn't want to change the logic too much.

The original code was ...

https://github.com/jetty/jetty.project/blob/94beede357ec97942812d16cac250ded768dd325/jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/HostPort.java#L180-L184

Which is exactly the same, but using 0 instead of -1.